### PR TITLE
chore: bump to nixpkgs 23.11

### PR DESCRIPTION
**Pull request by tomberek**
## Proposed Changes

bump to 23.11

## Release Notes

N/A

### CODE REVIEW

**Comment by mkenigs** [comment]
We would also need to change:
https://github.com/flox/flox/blob/b4b14e86229843bb68462e00425d5939c82fbded/pkgdb/src/registry.cc#L381

**Comment by stahnma** [comment]
@tomberek I do think we should do this. Should we just coordinate a day, announce in slack and make it happen?

File: Justfile [hunk 1]
```diff
@@ -34,11 +34,11 @@ cargo_test_invocation := "PKGDB_BIN=${PKGDB_BIN} cargo test --workspace"
 
 # Build the compilation database
 build-cdb:
-    @make -C pkgdb -j -s cdb
+    @make -C pkgdb -j 4 -s cdb
```
Comment (stahnma) [inline Justfile:37]: How is this related to changing the version of nixpkgs we're tracking?

Comment (tomberek) [reply Justfile]: It isn't, it crept in as a fix to avoid OOM-ing. removing....

File: cli/tests/package-search.bats [hunk 2]
```diff
@@ -81,15 +81,15 @@ setup_file() {
 
 @test "'FLOX_FEATURES_SEARCH_STRATEGY=match flox search' expected number of results: 'hello'" {
   FLOX_FEATURES_SEARCH_STRATEGY=match run --separate-stderr "$FLOX_BIN" search hello --all
-  assert_equal "${#lines[@]}" 11
```
Comment (ysndr) [inline cli/tests/package-search.bats:84]: Are these tests really sensitive to the nixpkgs we build flox with?
I thought of those to be basically _some independent_ rev.

would be better to isolate them a bit more so we can do future updates without changing lots of tests that depend on specific _locked_ results...

Comment (tomberek) [reply cli/tests/package-search.bats:84]: Yes, they are sensitive to the nixpkgs version. One possibility is keep testing on the same nixpkgs as before. That might be enough, see revision f44164dfabbd207cb27e39e82bb8f6c6dd8cb631 (two commits back). This passes all tests. @mkenigs I forgot to mention this, it could be a reasonable stepping stone before trying to change the tests.

**Comment by mkenigs** [comment]
Work rebased and merged in https://github.com/flox/flox/pull/828
